service/switchboardapp/src/main/java/com/switchboard/app/domain/Encoder.java

```diff
@@ -0,0 +1,24 @@
+package com.switchboard.app.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.*;
+
+import javax.persistence.*;
+
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter@Setter
+@JsonIgnoreProperties({"hibernateLazyIntializer","handler","device"})
+@ToString
+public class Encoder {
+
+    @Id
+    @Column(name = "serial_number")
+    private Long serialNumber;
```

Comment:
Add @NotNull unless that's already validated by the other annotations. If you do, add @Valid to the create encoder endpoint Encoder argument.

Reply:
fixed